src/Endpoints/Delegates/HandlesDocuments.php

```diff
@@ -139,9 +141,19 @@ public function deleteDocument($documentId): array
         return $this->http->delete(self::PATH.'/'.$this->uid.'/documents/'.$documentId);
     }
 
-    public function deleteDocuments(array $documents): array
+    public function deleteDocuments(array $options): array
     {
-        return $this->http->post(self::PATH.'/'.$this->uid.'/documents/delete-batch', $documents);
+        try {
+            if (\array_key_exists('filter', $options) && $options['filter']) {
+                return $this->http->post(self::PATH.'/'.$this->uid.'/documents/delete', $options);
+            }
+
+            // backwards compatibility:
+            // expect to be a array to send alongside as $documents_ids.
+            return $this->http->post(self::PATH.'/'.$this->uid.'/documents/delete-batch', $options);
+        } catch (InvalidResponseBodyException $e) {
+            throw ApiException::rethrowWithHint($e, __FUNCTION__);
```

Comment:
I don't know exactly what it is `ApiException`
If the request is sent to v.1.1 the return will be a 405 `Method Not Allowed` so it would be a `ComunicationError` instead.
Isn't there an error plus something more general? Because the day we have a new field that is not a filter in a future version, it will be followed by a possible `ApiException`.
Maybe it would be good to time do it now but it's up to you

Reply:
I tried locally by using a different version in the docker-compose, and that was the error I got. To improve this I think I could catch any error and give always give the hint.

Reply:
`ApiException` is just a class I used to put the method `rethrowWithHint` on it. Nothing special there, but I choose it because it is the closest class I have to a `BaseError` class or something

Reply:
I have reverified in all my SDKs and a 405 returns a `MeilisearchAPIError` in my SDKs not a `CommunicationError`.
My bad